protos/mocap/mocap.proto

```diff
@@ -0,0 +1,162 @@
+    rpc SetOdometry(SetOdometryRequest) returns(SetOdometryResponse) {}
+}
+
+message SetVisionPositionEstimateRequest {
+    VisionPositionEstimate vision_position_estimate = 1;
+}
+message SetVisionPositionEstimateResponse {
+    MocapResult mocap_result = 1;
+}
+
+message SetAttitudePositionMocapRequest {
+    AttitudePositionMocap attitude_position_mocap = 1;
+}
+message SetAttitudePositionMocapResponse {
+    MocapResult mocap_result = 1;
+}
+
+message SetOdometryRequest {
+    Odometry odometry = 1;
+}
+message SetOdometryResponse {
+    MocapResult mocap_result = 1;
+}
+
+// Global position/attitude estimate from a vision source.
+message VisionPositionEstimate {
+    uint64 time_usec = 1; // PositionBody frame timestamp UNIX Epoch time (0 to use Backend timestamp)
+    PositionBody position_body = 2; // Global position (m)
+    AngleBody angle_body = 3; // Body angle (rad).
+    Covariance pose_covariance = 4; // Pose cross-covariance matrix.
+}
+
+// Motion capture attitude and position
+message AttitudePositionMocap {
+    uint64 time_usec = 1; // PositionBody frame timestamp UNIX Epoch time (0 to use Backend timestamp)
+    Quaternion q = 2; // Attitude quaternion (w, x, y, z order, zero-rotation is 1, 0, 0, 0)
+    PositionBody position_body = 3; // Body Position (NED)
+    Covariance pose_covariance = 4; // Pose cross-covariance matrix.
+}
+
+// Odometry message to communicate odometry information with an external interface.
+message Odometry {
+    // Mavlink frame id
+    enum MavFrame {
+        STUB = 0; // Stub for Protobuf
+        BODY_FRD = 12; // Body fixed frame of reference, Z-down (x: forward, y: right, z: down). Replacement for MAV_FRAME_BODY_NED, MAV_FRAME_BODY_OFFSET_NED.
+        BODY_FLU = 13; // Body fixed frame of reference, Z-up (x: forward, y: left, z: up).
+        LOCAL_FRD = 20; // Forward, Right, Down coordinate frame. This is a local frame with Z-down and arbitrary F/R alignment (i.e. not aligned with NED/earth frame). Replacement for MAV_FRAME_MOCAP_NED, MAV_FRAME_VISION_NED, MAV_FRAME_ESTIM_NED.
+        LOCAL_FLU = 21; // Forward, Left, Up coordinate frame. This is a local frame with Z-up and arbitrary F/L alignment (i.e. not aligned with ENU/earth frame). Replacement for MAV_FRAME_MOCAP_ENU, MAV_FRAME_VISION_ENU, MAV_FRAME_ESTIM_ENU.
+    }
+
+    // Enumeration of estimator types
+    enum MavEstimatorType {
+        UNKNOWN = 0; // Unknown type of the estimator.
+        NAIVE = 1; // This is a naive estimator without any real covariance feedback.
+        TYPE_VISION = 2; // Computer vision based estimate. Might be up to scale.
+        VIO = 3; // Visual-inertial estimate.
+        GPS = 4; // Plain GPS estimate.
+        INS = 5; // Estimator integrating GPS and inertial sensing.
+        MOCAP = 6; // Estimate from external motion capturing system.
+        LIDAR = 7; // Estimator based on lidar sensor input.
+        AUTOPILOT = 8; // Estimator on autopilot.
+    }
```

Comment:
At this time field `estimator_type` does not used in PX4.
But I think that ability to use it in default plugin will be helpful for MoCap and VIO development.

Reply:
I'm not sure about that. If it is not used, it means that we expose something that will fail. I would say that we only publish things that work with an actual autopilot. For development, one can always use a branch, and merge that into master when it is working.

In other words, I would be in favor of having in our proto files only things that we can test against an actual autopilot. We can always extend later.

What do you think?

Reply:
I very much agree. Otherwise we end up with the same problems as MAVLink where you don't know what to set for many params and you don't know if they are even used, supported, or what effects they have.

Reply:
Good clarification. I think you are right.